extensions/amp-story-interactive/amp-story-interactive.md

```diff
@@ -1,10 +1,310 @@
+
+While the component by default adapts to the screen size with the variable font-size, it doesn't stay perfectly consistent across screen sizes. It's possible to use the aspect-ratio layer in order to create layouts that will scale perfectly with different screen sizes, by setting the font-size in ems on the component.
+
+The width can be set either in ems or percentages of the parent width, and it will behave perfectly consistent (while keeping it between the min and max widths).
+
+[sourcecode:html]
+<amp-story-grid-layer template="fill" aspect-ratio="400:600">
+
+  <div class="center-quiz">
+    <amp-story-interactive-quiz
+      style="font-size:0.2em"
+      ...>
+    </amp-story-interactive-quiz>
+  </div>
+</amp-story-grid-layer>
+[/sourcecode]
+
+### Adaptive font sizes for options
+
+Polls and binary polls can adapt their font-sizes according to the content types. When they are created, depending on the number of lines that they have for the option text, they can increase or decrease the font size. The font-size will default to the largest possible that accommodates all the texts being measured. For instance, if one of the options requires a small font-size (because it has a lot of text) but the other options don't, the small font-size will be used for all the texts involved. Note that this only optimizes the content of the components to fit better, but they don't change the inherent size of the components.
+
+Binary polls only on post-selection will show different font-sizes for the options. They will hold a large font-size if all the options are either emojis or have short texts (length <= 3). Medium font-size will be used if at least one of the options are longer than 3 characters but both can be displayed in one line; and small font-size will be used if at least one option has two lines.
+
+Polls will also adapt the font-size of the options depending on the content. If all the options fit in one line, they will use a large font-size. If any of the options require 2 lines, all the options will be reduced in size to accommodate the option text in the chip.
+
+For a live demo, check out the [binary-poll](https://codepen.io/mszylkowski/pen/oNxogoV) and [poll](https://codepen.io/mszylkowski/pen/ZEWaBoZ) Codepens, and change the option texts (be sure to select "Answered" on the binary-poll demo to see the size change).
+
+## Animations
+
+The designs provide a subtle animation that enhances the interactivity aspect of the component, and are also animated when an option is selected to show the aggregate data.
+To add entering animations, [amp-story animations](https://amp.dev/documentation/components/amp-story/?format=stories#animation-attributes) can be added to all components.
+
+## Analytics
+
+The component comes with support for [amp-analytics](https://amp.dev/documentation/components/amp-analytics/) events out of the box with the event `story-interactive`, and will report when an option is selected:
```

Comment:
The `amp-story-interactive` component elements support [`amp-analytics`](https://amp.dev/documentation/components/amp-analytics/). Report a selected option by adding the `story-interactive` event to your configuration: